### PR TITLE
Add configuration to disable forking

### PR DESCRIPTION
**Pull request by ArulselvanMadhavan**
My attempt to work on https://github.com/fthomas/scala-steward/issues/197

Submitting for initial feedback.

Working on unit tests.

### CODE REVIEW

**Comment by ArulselvanMadhavan** [comment]
Hi @fthomas ,

When you have some time, can you look into this PR and let me know your thoughts?

Thanks.

File: project/plugins.sbt [hunk 1]
```diff
@@ -9,3 +9,4 @@ addSbtPlugin("org.portable-scala" % "sbt-scalajs-crossproject" % "0.6.0")
 addSbtPlugin("org.scoverage" % "sbt-scoverage" % "1.5.1")
 addSbtPlugin("org.tpolecat" % "tut-plugin" % "0.6.10")
 addSbtPlugin("com.typesafe.sbt" % "sbt-native-packager" % "1.3.15")
+addSbtPlugin("ch.epfl.scala" % "sbt-bloop" % "1.1.2")
```
Comment (Philippus) [inline project/plugins.sbt]: out of curiosity: why was this added?

**Comment by Philippus** [comment]
I tried this out, it worked for me!
Maybe add some information and the new config parameter to the `README.md` so it is clear it is an option.

File: project/plugins.sbt (hunk 1 above)
Comment (ArulselvanMadhavan) [reply project/plugins.sbt]: I accidentally checked it in. I use `bloop` while developing. So, I added it and forgot to remove it.

File: modules/core/src/main/scala/org/scalasteward/core/dependency/DependencyService.scala [hunk 2]
```diff
@@ -58,7 +59,7 @@ class DependencyService[F[_]](
       _ <- logger.info(s"Refresh dependencies of ${repo.show}")
       cloneUrl = util.uri.withUserInfo(repoOut.clone_url, config.gitHubLogin)
       _ <- gitAlg.clone(repo, cloneUrl)
-      parent <- repoOut.parentOrRaise[F]
+      parent <- repoOut.parentOrRaise[F](config)
       parentCloneUrl = util.uri.withUserInfo(parent.clone_url, config.gitHubLogin)
       _ <- gitAlg.syncFork(repo, parentCloneUrl, parent.default_branch)
```
Comment (fthomas) [inline modules/core/src/main/scala/org/scalasteward/core/dependency/DependencyService.scala]: In https://github.com/fthomas/scala-steward/pull/208/files#diff-bdb025c4edf5aaff2f7f02c1a209ad42R65 `syncFork` is guarded with `doNotFork`. Why not here?

File: modules/core/src/main/scala/org/scalasteward/core/github/data/RepoOut.scala [hunk 3]
```diff
@@ -30,8 +31,9 @@ final case class RepoOut(
     clone_url: Uri,
     default_branch: Branch
 ) {
-  def parentOrRaise[F[_]](implicit F: ApplicativeThrowable[F]): F[RepoOut] =
-    parent.fold(F.raiseError[RepoOut](new Throwable(s"repo $name has no parent")))(F.pure)
+  def parentOrRaise[F[_]](config: Config)(implicit F: ApplicativeThrowable[F]): F[RepoOut] =
```
Comment (fthomas) [inline modules/core/src/main/scala/org/scalasteward/core/github/data/RepoOut.scala]: This needs a better name now.

File: modules/core/src/main/scala/org/scalasteward/core/nurture/NurtureAlg.scala [hunk 4]
```diff
@@ -56,13 +56,14 @@ class NurtureAlg[F[_]](
...
-      repoOut <- gitHubApiAlg.createFork(repo)
-      parent <- repoOut.parentOrRaise[F]
+      repoOut <- gitHubApiAlg.createOrGetRepoInfo(config, repo)
+      parent <- repoOut.parentOrRaise[F](config)
       cloneUrl = util.uri.withUserInfo(repoOut.clone_url, config.gitHubLogin)
       parentCloneUrl = util.uri.withUserInfo(parent.clone_url, config.gitHubLogin)
       _ <- gitAlg.clone(repo, cloneUrl)
       _ <- gitAlg.setAuthor(repo, config.gitAuthor)
-      _ <- gitAlg.syncFork(repo, parentCloneUrl, parent.default_branch)
+      _ <- if (config.doNotFork) F.pure(Unit)
```
Comment (fthomas) [inline modules/core/src/main/scala/org/scalasteward/core/nurture/NurtureAlg.scala]: I think you can use `F.unit` instead of `F.pure(Unit)`.

File: modules/core/src/main/scala/org/scalasteward/core/github/GitHubApiAlg.scala [hunk 5]
```diff
@@ -47,14 +48,32 @@ trait GitHubApiAlg[F[_]] {
...
+    else createFork(repo)
+
+  def createOrGetRepoInfoWithBranchInfo(
+      config: Config,
+      repo: Repo
+  )(implicit F: MonadThrowable[F]): F[(RepoOut, BranchOut)] =
+    if (config.doNotFork)
+      getRepoInfo(repo).flatMap(
+        repoOut => getDefaultBranch(repoOut).map(branchOut => repoOut -> branchOut)
+      )
```
[the remark this answers is not in the document]
Comment (ArulselvanMadhavan) [reply modules/core/src/main/scala/org/scalasteward/core/github/GitHubApiAlg.scala]: Agreed. :+1:

File: modules/core/src/main/scala/org/scalasteward/core/github/GitHubApiAlg.scala [hunk 6]
```diff
@@ -47,14 +48,32 @@ trait GitHubApiAlg[F[_]] {
   ): F[List[PullRequestOut]]
 
   def createForkAndGetDefaultBranch(
+      config: Config,
       repo: Repo
   )(implicit F: MonadThrowable[F]): F[(RepoOut, BranchOut)] =
     for {
       fork <- createFork(repo)
-      parent <- fork.parentOrRaise[F]
+      parent <- fork.parentOrRaise[F](config)
```
[the remark this answers is not in the document]
Comment (ArulselvanMadhavan) [reply modules/core/src/main/scala/org/scalasteward/core/github/GitHubApiAlg.scala]: 👍 Removed

File: modules/core/src/main/scala/org/scalasteward/core/dependency/DependencyService.scala (hunk 2 above)
Comment (ArulselvanMadhavan) [reply modules/core/src/main/scala/org/scalasteward/core/dependency/DependencyService.scala]: Thanks! Added a guard.

**Comment by ArulselvanMadhavan** [comment]
I addressed all of the PR suggestions. Working on the final unit test

**Comment by ArulselvanMadhavan** [comment]
I managed to get the tests pass the coverage thresholds.

File: project/plugins.sbt (hunk 1 above)
Comment (ArulselvanMadhavan) [reply project/plugins.sbt]: @fthomas Is it Ok if I leave this plugin in there ?

File: modules/core/src/test/scala/org/scalasteward/core/MockState.scala [hunk 7]
```diff
@@ -18,4 +20,21 @@ object MockState {
 
   def empty: MockState =
     MockState(Vector.empty, Vector.empty)
+
+  // Unable to make this implicit. Ambiguous implicit error
```
Comment (ArulselvanMadhavan) [inline modules/core/src/test/scala/org/scalasteward/core/MockState.scala]: I am unable to make this implicit. If @fthomas you have any suggestions, let me know. If not, I may have to look into this later with a fresh mind.

Comment (fthomas) [reply modules/core/src/test/scala/org/scalasteward/core/MockState.scala]: I think it would be easier to change `MockEnv` to something that is already a `MonadThrowable`. But we should do this later. For now I'm fine with your current solution.

**Comment by fthomas** [comment]
This LGTM, thanks @ArulselvanMadhavan! I like that this just changes the original code in places where steward forks repos, syncs repos, and uses the GitHub login for creating PRs. :+1: Also many thanks for the increase of test coverage!

File: modules/core/src/test/scala/org/scalasteward/core/MockState.scala (hunk 7 above)
Comment (fthomas) [reply modules/core/src/test/scala/org/scalasteward/core/MockState.scala]: #274 changes `MockEnv[A]` from `State[MockState, A]` to `StateT[IO, MockState, A]` so that we get the `MonadError` instance for free and this explicit instance isn't necessary anymore.

**Comment by fthomas** [comment]
@all-contributors please add @ArulselvanMadhavan for code, test
